.github/workflows/ci.yml

```diff
@@ -71,7 +71,7 @@ jobs:
         run: rustup update stable
       - uses: Swatinem/rust-cache@v1
       - name: Install cargo-hack
-        run: cargo install cargo-hack --version 0.5.26
+        run: cargo install cargo-hack --version 0.3.0
```

Comment:
I would recommend using install-action like master does.

https://github.com/tokio-rs/tokio/blob/65e7715909694d8d7d15b081cc3790b4ea1ad300/.github/workflows/ci.yml#L165-L168

Reply:
I replaced:
```
run: cargo install cargo-hack --version 0.3.0
```
with
```
uses: taiki-e/install-action@v2 
with: 
  tool: cargo-hack@0.3.0 
```
some jobs failed. I believe the reason was this warning:
```
Warning: cargo-hack@0.3.0 for 'linux' is not supported; fallback to cargo-binstall
info: install-action does not support cargo-hack@0.3.0; fallback to cargo-binstall
```
So I decided to revert back to installing it using `cargo install`.

Reply:
> tool: cargo-hack@0.3.0 

Removing '@0.3.0' should fix the error.

Reply:
But I need the 0.3.0 version. It's the latest version that works.

Reply:
Why 0.3.0 is needed?

Reply:
To be clear: cargo-hack is also compatible with much older versions of cargo than required for installation.

https://github.com/taiki-e/cargo-hack?tab=readme-ov-file#from-source

> cargo-hack is usually runnable with Cargo versions older than the Rust version required for installation (e.g., `cargo +1.31 hack check`). Currently, to run cargo-hack requires Cargo 1.26+.

Reply:
You're right. I mistakenly thought the action compiles `cargo-hack`. I've made the necessary changes.